lib/LegacyUsers.php

```diff
@@ -422,7 +422,15 @@ public static function change_user_pass($db, array $details): bool
         }
 
         $query = "SELECT id, password FROM users WHERE username = {$db->quote($_SESSION ["userlogin"], 'text')}";
+        
         $response = $db->queryRow($query);
+        if ($response['password'] == 'LDAP_USER') {
```

Comment:
Instead of checking the password, it would be better to get the `use_ldap` field from the database and check if it's true in this case.

Reply:
Done. Now query got three attributes one of them is 'use_ldap'